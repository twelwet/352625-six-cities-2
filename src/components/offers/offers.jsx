import React from "react";
import PropTypes from "prop-types";
import Offer from "../../components/offer/offer.jsx";

const Offers = (props) => {
  const {cityOffers, onSelect, onUnselect} = props;
  return <div className="cities__places-list places__list tabs__content">
    {cityOffers.map((item, index) => {
      return (
        <Offer
          {...item}
          key={item.id}
          onOfferHover={() => onSelect(index)}
          onOfferLeave={() => onUnselect()}
        />
      );
    })}
  </div>;
};


Offers.propTypes = {
  cityOffers: PropTypes
    .arrayOf(PropTypes
      .exact({
        id: PropTypes.number.isRequired,
        city: PropTypes.exact({
          location: PropTypes.exact({
            name: PropTypes.string.isRequired,
            latitude: PropTypes.number.isRequired,
            longitude: PropTypes.number.isRequired,
            zoom: PropTypes.number.isRequired
          })
        }),
        location: PropTypes.exact({
          latitude: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          zoom: PropTypes.number.isRequired
        }),
        description: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        image: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        isPremium: PropTypes.bool.isRequired,
        isBookmark: PropTypes.bool.isRequired,
      })
    ),
  onSelect: PropTypes.func.isRequired,
  onUnselect: PropTypes.func.isRequired
};

export default Offers;
