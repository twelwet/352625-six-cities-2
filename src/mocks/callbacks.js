const descriptionClickHandler = (evt) => {
  evt.preventDefault();
  evt.currentTarget.style.color = `red`;
};

export {descriptionClickHandler};

