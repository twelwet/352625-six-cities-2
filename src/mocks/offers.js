const offers = [
  {
    id: 1,
    city: {
      location: {
        name: `Amsterdam`,
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 12
    },
    description: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    price: 120,
    image: `img/apartment-01.jpg`,
    rating: 100,
    isPremium: true,
    isBookmark: false
  },
  {
    id: 2,
    city: {
      location: {
        name: `Amsterdam`,
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 12
    },
    description: `Wood and stone place`,
    type: `Private room`,
    price: 80,
    image: `img/room.jpg`,
    rating: 50,
    isPremium: false,
    isBookmark: true,
  },
  {
    id: 3,
    city: {
      location: {
        name: `Amsterdam`,
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    location: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 12
    },
    description: `Canal View Prinsengracht`,
    type: `Apartment`,
    price: 132,
    image: `img/apartment-02.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 4,
    city: {
      location: {
        name: `Amsterdam`,
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    location: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 180,
    image: `img/apartment-03.jpg`,
    rating: 100,
    isPremium: true,
    isBookmark: false
  },
  {
    id: 5,
    city: {
      location: {
        name: `Amsterdam`,
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    location: {
      latitude: 52.369553943508,
      longitude: 4.939309666406198,
      zoom: 12
    },
    description: `Wood and stone place`,
    type: `Private room`,
    price: 80,
    image: `img/room.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 6,
    city: {
      location: {
        name: `Paris`,
        latitude: 48.864716,
        longitude: 2.349014,
        zoom: 12
      }
    },
    location: {
      latitude: 48.86555432,
      longitude: 2.34856278,
      zoom: 12
    },
    description: `Wood and stone place`,
    type: `Private room`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 7,
    city: {
      location: {
        name: `Paris`,
        latitude: 48.864716,
        longitude: 2.349014,
        zoom: 12
      }
    },
    location: {
      latitude: 48.860955394,
      longitude: 2.348687458,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 8,
    city: {
      location: {
        name: `Cologne`,
        latitude: 50.941357,
        longitude: 6.958307,
        zoom: 12
      }
    },
    location: {
      latitude: 50.9427634598,
      longitude: 6.9683475654,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 9,
    city: {
      location: {
        name: `Brussels`,
        latitude: 50.85045,
        longitude: 4.34878,
        zoom: 12
      }
    },
    location: {
      latitude: 50.8517845976,
      longitude: 4.3463294857,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 10,
    city: {
      location: {
        name: `Brussels`,
        latitude: 50.85045,
        longitude: 4.34878,
        zoom: 12
      }
    },
    location: {
      latitude: 50.85254739,
      longitude: 4.34375675,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 11,
    city: {
      location: {
        name: `Hamburg`,
        latitude: 53.551086,
        longitude: 9.993682,
        zoom: 12
      }
    },
    location: {
      latitude: 53.5537409586,
      longitude: 9.9992837456,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
  {
    id: 12,
    city: {
      location: {
        name: `Dusseldorf`,
        latitude: 51.233334,
        longitude: 6.783333,
        zoom: 12
      }
    },
    location: {
      latitude: 51.2329387456,
      longitude: 6.7830495867,
      zoom: 12
    },
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    price: 80,
    image: `img/apartment-01.jpg`,
    rating: 80,
    isPremium: false,
    isBookmark: false
  },
];

export {offers};
