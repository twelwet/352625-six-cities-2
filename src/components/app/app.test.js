import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";
import {offers} from "../../mocks/offers.js";
import {onDescriptionClick} from "../../mocks/callbacks.js";

it(`App correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<App
      offers={offers}
      onDescriptionClick={onDescriptionClick}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
