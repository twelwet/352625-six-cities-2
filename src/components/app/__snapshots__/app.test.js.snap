// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/offer/:id"
    render={[Function]}
  />
</Switch>
`;
